internal/pkg/errorconcept/db.go

```diff
@@ -36,7 +37,12 @@ func (r dbNotFound) httpErrorCode() int {
 }
 
 func (r dbNotFound) isA(err error) bool {
-	return err == db.ErrNotFound
+	if err == db.ErrNotFound {
+		return true
+	} else {
```

Comment:
This else block may be best split out into its own ErrorConceptType. With the addition of the else block, each place which performs error-type-checking via string, will include this extra cycle. There were no scenarios that stood out to me to suggest that this may break anything, however, I think its best to split this out into its own ErrorConceptType and err of the side of caution.

As a side note, the pattern that is in place will help spotlight error-handling inconsistencies throughout edgex-go. This case here is a good example of that. With it defined as its own ErrorConceptType, it helps easily identify the different DBNotFound errors that we check (those which are operator bound and those which are not).

After splitting the err.(data.ErrDbNotFound) type out, I will approve.